### PR TITLE
Trees

### PR DESCRIPTION
**Pull request by dlewburg**
- [ ] Top-level README “Table of Contents” is updated
 - [ ] README for this challenge is complete
       - [ ] Summary, Description, Approach & Efficiency, Solution
       - [ ] Picture of whiteboard
       - [ ] Link to code
 - [x] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
       - [ ] “Happy Path” - Expected outcome
       - [x] Expected failure
       - [x] Edge Case (if applicable/obvious)

### CODE REVIEW

File: java/datastructures/trees/lib/src/main/java/trees/BinaryTree.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    } else {
+      if (current.getRight() == null) {
+        current.setRight(new Node(value));
+      } else {
+        insertHelper(current.getRight(), value);
+      }
+    }
+  }
+
+  public int[] preOrder(List<Integer> traversal) {
```
Comment (lucy-gelderloos) [inline java/datastructures/trees/lib/src/main/java/trees/BinaryTree.java]: Here, you're passing in a List called `traversal`, but then creating a new list called `result`. The method then adds all of the traversal values to `result`, so `traversal` is never changed. See also your inOrder and postOrder methods. Once I fixed this issue, all your tests passed!
